### ISSUE TITLE
Remove deprecated twisted.persisted.journal

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4805]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4805|
|Type|task|
|Created|2011-01-15 00:49:52Z|
|Branch|https://github.com/twisted/twisted/tree/remove-twisted-persisted-journal-4805-2|

This has been deprecated since 10.2+, let's remove it. See [#4298](https://github.com/twisted/twisted/issues/4298).

<details><summary>Searchable metadata</summary>

```
trac-id__4805 4805
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_remove_twisted_persisted_journal_4805_2 branches/remove-twisted-persisted-journal-4805-2
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1295052592000000 1295052592000000
changetime__1331392949000000 1331392949000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

(In [32214]) Branching to 'remove-twisted-persisted-journal-4805'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

Whoops, this was re-deprecated in 11.0, since there was no newsfile in 10.1.  Can't delete it until 11.2.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [33664]) Branching to 'remove-twisted-persisted-journal-4805-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [33665]) remove deprecated t.persisted.journal, refs [#4805](https://github.com/twisted/twisted/issues/4805)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-twisted-persisted-journal-4805-2).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for working on this, thijs. Looks good, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33674]) Merge remove-twisted-persisted-journal-4805-2: twisted.persisted.journal is removed, deprecated since Twisted 11.0.

Author: thijs
Reviewer: jesstess
Fixes: [#4805](https://github.com/twisted/twisted/issues/4805)
